### PR TITLE
Avoid string property access in library_webgpu.js. NFC

### PR DESCRIPTION
**Pull request by sbc100**
I believe this was done as a defense against closure minification but I would hope that is not necessary today.

I checks several of these properties and they all appear to be present in `src/closure-externs/webgpu-externs.js`.

### CODE REVIEW

**Comment by kainino0x** [comment]
I did add these to the externs, but I hadn't gotten around to this yet - thanks!
The thing that held me back was that I don't know if Closure is able to analyze the types enough to apply the externs, for example in `WebGPU.mgrBuffer.get(bufferId).destroy();` does it know that `WebGPU.mgrBuffer.get(bufferId)` is a `GPUBuffer`? And if not then how does it know not to minify these things - does it just avoid minifying every identifier it sees in externs, if the type is not known?

And of course we don't have good test coverage so I could only verify that closure builds work for a few of the entry points. I don't know how much risk there is of the minification breaking on some and not others.

**Comment by sbc100** [comment]
I see.  Sounds like we need to some more test coverage.  I wonder how much we would need to confirm this won't break anyone?

**Review by kainino0x — APPROVED** [review]
I suppose I would be happy with some spot testing that it's not minifying things it shouldn't. The tests we have now should be enough, and I'm guessing you already tested those. I really want to have an understanding of how Closure actually figures this out, but if the tests pass and given it seems to work fine for other stuff in Emscripten, it must work somehow.

LGTM if the tests pass locally.

**Comment by kainino0x** [comment]
I thought we couldn't run the browser tests on CI because they require a GPU?
They should compile, but Closure minification errors won't be caught until the particular minified code is actually executed.

**Comment by kainino0x** [comment]
I am trying to reproduce the problem so I can make sure that it is caught by the tests.
I am easily able to reproduce the issue if I delete (most of) the webgpu externs and build my https://github.com/kainino0x/webgpu-cross-platform-demo sample (`.draw` becomes `.Sa`).
But if I just compile `test/webgpu_basic_rendering.cpp`, it does not reproduce. I am not sure yet why this is the case. It must have something to do with how it's built (using CMake).

**Comment by kainino0x** [comment]
There is somehow a difference between `SIMPLE` and `ADVANCED` mode in Closure. In Emscripten this is controlled by `-O2` vs `-O3`. I am not sure why I had such a hard time reproducing this. The results seem to keep varying.
- Previously I tried reproducing with O3 (copying the args from the cmake project) and it didn't repro.
- I was on this branch, and it reproduced with O3 but not with O2?
- I merged in `main`, and it started reproducing with both O3 and O2.
- I undid that merge, and now it still reproduces with both O3 and O2.

I am probably just messing up some cache thing. But I am pretty certain `-O3` reproduces the issue more reliably so I'll just add a test variant that uses it to this PR. That catches the issue, so we can merge it!

Side note, I did figure out the thing I was wondering about, how it "knew" that `draw` needed to be unminified if it didn't know the type of the object it was being called on. As I suspected, if I delete the extern for `GPURenderPassEncoder.prototype.draw` but keep the extern for `GPURenderBundleEncoder.prototype.draw`, the former still isn't minified. So it must be the case that if it doesn't know the type, then it assumes it could be some extern called `draw`, and avoids minifying it.

**Comment by kainino0x** [comment]
> In Emscripten this appears to be controlled by `-O2` vs `-O3`?

Well, I thought so, but maybe by `-O1` vs `-O2`.
Regardless, `SIMPLE` doesn't minify `draw` when the extern is missing and `ADVANCED` does.

File: test/test_browser.py [hunk 1]
```diff
@@ -4690,6 +4690,7 @@ def test_webgl_simple_extensions(self, simple_enable_extensions, webgl_version):
   @parameterized({
     'default': ([],),
     'closure': (['-sASSERTIONS', '--closure=1'],),
```
Comment (sbc100) [inline test/test_browser.py:4685]: The main run that we do of closure compiler always uses `ADVANCED_OPTIMIZATIONS` IIUC.

See https://github.com/emscripten-core/emscripten/blob/e0bd10a0b61e08d1ed30966b6d8351ea06376fd4/tools/link.py#L2218.

Here the closure_compiler() method is called without the `advanced` param, which means `advanced` defaults to `True`.

File: test/test_browser.py [hunk 2]
```diff
@@ -4690,6 +4690,7 @@ def test_webgl_simple_extensions(self, simple_enable_extensions, webgl_version):
...
+    'closure_advanced': (['-sASSERTIONS', '--closure=1', '-O3'],),
```
Comment (sbc100) [inline test/test_browser.py:4686]: I'm not sure we need both `-O0` and `-O3` here.

File: test/test_browser.py (hunk 1 above)
Comment (sbc100) [reply test/test_browser.py:4685]: Indeed if you build with `-O0` and `--closure=1` you can see (by adding -v) that it uses `--compilation_level ADVANCED_OPTIMIZATIONS`

Comment (kainino0x) [reply test/test_browser.py:4685]: I am not sure how this works internally, but the output with `-O1` looks like the output with `SIMPLE` and the output with `-O2` looks like the output with `ADVANCED`. Maybe there's a more fine-grained option than SIMPLE vs ADVANCED that is causing this?

The externs seem to sometimes affect both builds, but more reliably affect the ADVANCED build.

File: test/test_browser.py (hunk 2 above)
Comment (kainino0x) [reply test/test_browser.py:4686]: There are a few other tests that test closure with and without O3 so I was guessing there must be some expected difference:
https://github.com/emscripten-core/emscripten/blob/d278e2d0a0b9a86a880dfa57b3794721e0a3ccf4/test/test_other.py#L9650-L9656
Though I guess that could just be trying to change the optimization level of the Wasm.

Regardless in practice it does seem to make a difference. Though as mentioned it somehow seemed nondeterministic.

Comment (sbc100) [reply test/test_browser.py:4686]: Sounds good to me to land this as is then.

The nondeterministic part is certainly very worrying though.  If you can share a concrete example of that I would love to to see it.. there should not be any non-determinism in the toolchain clearly.

**Comment by sbc100** [comment]
I'm looking into the (unrelated) CI failures now.  I will rebase this once they are fixed.

**Pull request merged**